components/GameEngine/include/TurnManager.h

```diff
@@ -11,111 +11,93 @@
 #include "GameLooptyLoop.h"
 #include "Player.h"
 #include "SFML/Graphics.hpp"
-#include <algorithm>
 #include <memory>
 #include <vector>
 
-using namespace std;
 using namespace sf;
+using namespace std;
 
+/**
+ * @brief Manages the turn order and turn-based interactions in the game.
+ */
 class TurnManager : public LogObservable {
 private:
-    vector<shared_ptr<Character>> characterList;
-    int currentIndex = 0;
-    shared_ptr<Character> currentPlayer;
+    vector<shared_ptr<Character>> characterList;     ///< List of all characters in turn order.
+    int currentIndex = 0;                            ///< Index of the current character's turn.
+    shared_ptr<Character> currentPlayer;             ///< Pointer to the current player character.

```

Comment:
Maybe we should use the pointer to the player in the Campaign therefore having only 1 way to access the player.